tests/tpch/test_dask.py

```diff
@@ -645,3 +645,48 @@ def test_query_14(client, dataset_path, fs):
     )
 
     return pd.DataFrame({"promo_revenue": [result]})
+
+
+@pytest.mark.shuffle_p2p
+def test_query_17(client, dataset_path, fs):
+    """
+    select
+        round(sum(l_extendedprice) / 7.0, 2) as avg_yearly
+    from
+        lineitem,
+        part
+    where
+        p_partkey = l_partkey
+        and p_brand = 'Brand#23'
+        and p_container = 'MED BOX'
+        and l_quantity < (
+            select
+                0.2 * avg(l_quantity)
+            from
+                lineitem
+            where
+                l_partkey = p_partkey
+        )
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    avg_qnty_by_partkey = (
+        lineitem.groupby("l_partkey")
+        .l_quantity.mean()
+        .to_frame()
+        .rename(columns={"l_quantity": "l_quantity_avg"})
+    )
```

Comment:
This query might be faster if aggregated the joined and pre-filtered DF of lineitems and parts instead. Happy to evaluate that in a follow-up though.